src/3D/L3DSubMesh.cpp

```diff
@@ -122,7 +122,53 @@ bool L3DSubMesh::Load(const l3d::L3DFile& l3d, uint32_t meshIndex)
 		for (uint32_t j = 0; j < primitive.numTriangles * 3; j++)
 			indices[startIndex + j] = indexSpan[startIndex + j] + startVertex;
 
-		_primitives.emplace_back(Primitive {primitive.skinID, startIndex, primitive.numTriangles * 3});
+		struct MaterialTypeLutEntry
+		{
+			bool depthWrite;
+			bool alphaTest;
+			L3DSubMesh::Primitive::BlendMode blend;
+			bool modulateAlpha;  ///< Multiply ouput alpha by a uniform
+			bool thresholdAlpha; ///< Dismiss fragments below a certain threshold
+		};
+		static const std::array<MaterialTypeLutEntry, static_cast<uint32_t>(l3d::L3DMaterial::Type::_Count)> materialTypeLut = {
+		    {
+		        {true, false, L3DSubMesh::Primitive::BlendMode::Disabled, false, false},  // Smooth
+		        {true, false, L3DSubMesh::Primitive::BlendMode::Standard, false, false},  // SmoothAlpha
+		        {true, false, L3DSubMesh::Primitive::BlendMode::Disabled, false, false},  // Textured
+		        {true, false, L3DSubMesh::Primitive::BlendMode::Standard, true, false},   // TexturedAlpha
+		        {true, false, L3DSubMesh::Primitive::BlendMode::Standard, false, false},  // AlphaTextured
+		        {true, false, L3DSubMesh::Primitive::BlendMode::Standard, true, false},   // AlphaTexturedAlpha
+		        {false, false, L3DSubMesh::Primitive::BlendMode::Standard, true, false},  // AlphaTexturedAlphaNz
+		        {false, false, L3DSubMesh::Primitive::BlendMode::Standard, false, false}, // SmoothAlphaNz
+		        {false, false, L3DSubMesh::Primitive::BlendMode::Standard, true, false},  // TexturedAlphaNz
+		        {true, true, L3DSubMesh::Primitive::BlendMode::Standard, false, true},    // TexturedChroma
+		        {true, true, L3DSubMesh::Primitive::BlendMode::Additive, true, true},     // AlphaTexturedAlphaAdditiveChroma
+		        {false, true, L3DSubMesh::Primitive::BlendMode::Additive, true, true},    // AlphaTexturedAlphaAdditiveChromaNz
+		        {true, false, L3DSubMesh::Primitive::BlendMode::Additive, true, false},   // AlphaTexturedAlphaAdditive
+		        {false, false, L3DSubMesh::Primitive::BlendMode::Additive, true, false},  // AlphaTexturedAlphaAdditiveNz
+		        {0, 0, L3DSubMesh::Primitive::BlendMode::Disabled, 0, 0},                 // 0xe
```

Comment:
I'm trying to put the most "Undefined" value there as possible because I have no idea what that configuration is supposed to be. As far as I can tell no meshes use it and I haven't seen how vanilla reacts to those values.